packages/dev/core/src/Audio/sound.ts

```diff
@@ -905,9 +905,12 @@ export class Sound {
                     this.isPaused = false;
                     this._startTime = 0;
                     this._currentTime = 0;
-                    this._soundSource!.onended = () => void 0;
+                    if (this._soundSource) {
+                        this._soundSource.onended = () => void 0;
+                    }
+                    this._onended();
                 };
-                this._stopSoundSource(stopTime);
+                this._soundSource.stop(stopTime);

```

Comment:
This will break if _soundSource is empty ?

It also means we do not disconnect the source anymore ? (just wondering if intended)

Reply:
Ya, `_soundSource` is checked in the enclosing `else if` and it doesn't need to be disconnected until `play` is called again, so it's handled there, instead.